src/cli.ts

```diff
@@ -15,87 +16,166 @@ const defaultCommand = define({
       type: 'string',
       default: 'auto',
       description: `Package manager to use for packing ('auto' | 'npm' | 'yarn' | 'pnpm' | 'bun')`
+    },
+    'list-tarball-files': {
+      type: 'boolean',
+      default: false,
+      description: 'List all files in the tarball',
     }
   },
   async run(ctx) {
     const root = ctx.positionals[0];
     let pack = ctx.values.pack as PackType;
+    const showAllFiles = ctx.values['list-tarball-files'] as boolean;
 
     prompts.intro('Generating report...');
 
     if (typeof pack === 'string' && !allowedPackTypes.includes(pack)) {
       prompts.cancel(
         `Invalid '--pack' option. Allowed values are: ${allowedPackTypes.join(', ')}`
       );
-      process.exit(1);
+      throw new Error('Invalid --pack option');
     }
 
     // If a path is passed, see if it's a path to a file (likely the tarball file)
+    let isTarball = false;
     if (root) {
-      const stat = await fs.stat(root).catch(() => {});
-      const isTarballFilePassed = stat?.isFile() === true;
-      if (!isTarballFilePassed) {
+      try {
+        const stat = await fs.stat(root);
+        if (stat.isFile()) {
+          const buffer = await fs.readFile(root);
+          pack = {tarball: buffer.buffer};
+          isTarball = true;
+        } else if (stat.isDirectory()) {
+          // It's a directory, which is fine
```

Comment:
is it?

if we pass a positional arg, presumably its to a tgz we want to unpack

in all other cases, the pack option would be a package manager and we can just look for the associated node_modules

I think? 👀